libmeepgeom/meepgeom.cpp

```diff
@@ -676,10 +661,17 @@ bool geom_epsilon::get_material_pt(material_type &material, const meep::vec &r)
     else
       material = default_material;
   }
-  else if (md->which_subclass == material_data::MATERIAL_FUNCTION) {
-    // TODO figure this out
-    // material = eval_material_func(md->user_material_func,md->user_data,p);
+  else if (md->which_subclass == material_data::MATERIAL_FUNCTION) { 
+
+    // is *THIS* what I want to do here? This can't be right...
+    material_data *new_material_data = (material_data *)malloc(sizeof(material_data));
+    new_material_data->which_subclass = material_data::MEDIUM;
+    new_material_data->user_func = 0;
+    new_material_data->user_data = 0;
+    new_material_data->medium = md->user_func(p, md->user_data);
+    material.data = (void *)new_material_data;
```

Comment:
No, I think it should be something like:
```jl
material. which_subclass = material_data::MEDIUM;
material.medium = md->user_func(p, md->user_data);
destroy_material = true;
```

Reply:
The bit that I still can't figure out here is that `material_type_destroy` is an empty function in libctlgeom. It doesn't deallocate anything. So how are all of these `medium` structures newly allocated by `user_func` ever going to get freed up? Am I just overlooking something in all the flipping back and forth between hierarchies of data structures in different code files and different source trees?